src/libs/ApiUtils.ts

```diff
@@ -16,29 +16,25 @@ Environment.getEnvironment().then((envName) => {
     // We connect here, so we have the updated ENV_NAME when Onyx callback runs
     Onyx.connect({
         key: ONYXKEYS.USER,
-        callback: (val) => {
+        callback: (value) => {
             // Toggling between APIs is not allowed on production and internal dev environment
             if (ENV_NAME === CONST.ENVIRONMENT.PRODUCTION || CONFIG.IS_USING_LOCAL_WEB) {
                 shouldUseStagingServer = false;
                 return;
             }
 
             const defaultToggleState = ENV_NAME === CONST.ENVIRONMENT.STAGING || ENV_NAME === CONST.ENVIRONMENT.ADHOC;
-            shouldUseStagingServer = lodashGet(val, 'shouldUseStagingServer', defaultToggleState);
+            shouldUseStagingServer = value?.shouldUseStagingServer ?? defaultToggleState;
         },
     });
 });
 
 /**
  * Get the currently used API endpoint
  * (Non-production environments allow for dynamically switching the API)
- *
- * @param {Object} [request]
- * @param {Boolean} [request.shouldUseSecure]
- * @returns {String}
  */
-function getApiRoot(request) {
-    const shouldUseSecure = lodashGet(request, 'shouldUseSecure', false);
+function getApiRoot(request?: Pick<Request, 'shouldUseSecure'>): string {
```

Comment:
NAB: Why not just type `Request`?

Reply:
Does Request have a `shouldUseSecure` parameter?

Reply:
Fixed @blazejkustra @madmax330